agent/src/main/java/io/jenkins/plugins/remotingkafka/Agent.java

```diff
@@ -0,0 +1,78 @@
+package io.jenkins.plugins.remotingkafka;
+
+import io.jenkins.plugins.remotingkafka.kafkaclient.KafkaConsumerClient;
+import io.jenkins.plugins.remotingkafka.kafkaclient.KafkaProducerClient;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class Agent {
+    private static final Logger LOGGER = Logger.getLogger(Agent.class.getName());
+
+    private final Options options;
+
+    public Agent(Options options) {
+        this.options = options;
+        LOGGER.finest("Agent created with " + options);
```

Comment:
In such case you may expose secrets to Log once there is some security control for agent connection. If you do not plan to do filtering, better to not log it for now